pipeline-model-definition/src/main/groovy/org/jenkinsci/plugins/pipeline/modeldefinition/Utils.groovy

```diff
@@ -622,6 +624,12 @@ class Utils {
                     isJobChanged = true
                 }
 
+                DisableRestartFromStage declaredDisableRestartFromStageOption = (DisableRestartFromStage) rawOptions.find { it instanceof DisableRestartFromStage }
+                if(j.getAction(DisableRestartFromStageAction.class) == null && declaredDisableRestartFromStageOption != null){
+                    j.addAction(new DisableRestartFromStageAction())
+                } else if(j.getAction(DisableRestartFromStageAction.class) != null && declaredDisableRestartFromStageOption == null) {
+                    j.removeAction(j.getAction(DisableRestartFromStageAction.class))
```

Comment:
nit: I would use a local variable to avoid looking up the action more than once (untested):
 
```suggestion
                DisableRestartFromStageAction disableRestartFromStageAction = j.getAction(DisableRestartFromStageAction.class)
                if(disableRestartFromStageAction == null && declaredDisableRestartFromStageOption != null){
                    j.addAction(new DisableRestartFromStageAction())
                } else if(disableRestartFromStageAction != null && declaredDisableRestartFromStageOption == null) {
                    j.removeAction(disableRestartFromStageAction)
```

Reply:
Valid point. I'll do that

Reply:
Please verify the change